draft-packages/form/KaizenDraft/Form/Primitives/InputSearch/InputSearch.tsx

```diff
@@ -0,0 +1,97 @@
+import { Icon } from "@kaizen/component-library"
+import { LoadingSpinner } from "@kaizen/draft-loading-spinner"
+import classnames from "classnames"
+import React from "react"
+import search from "@kaizen/component-library/icons/search.icon.svg"
+import clear from "@kaizen/component-library/icons/clear.icon.svg"
+
+import styles from "../Input/styles.scss"
+
+export interface InputSearchProps
+  extends Omit<
+    React.InputHTMLAttributes<HTMLInputElement>,
+    "className" | "defaultValue"
+  > {
+  classNameAndIHaveSpokenToDST?: string
+  reversed?: boolean
+  loading?: boolean
+  secondary?: boolean
+  onClear?: () => void
+}
+
+const InputSearch: React.FunctionComponent<InputSearchProps> = (
+  props: InputSearchProps
+) => {
+  const {
+    value,
+    onChange,
+    onClear,
+    classNameAndIHaveSpokenToDST,
+    placeholder,
+    disabled = false,
+    reversed = false,
+    loading,
+    secondary,
+    ...genericInputProps
+  } = props
+
+  return (
+    <div
+      className={classnames(
+        styles.wrapper,
+        styles.withStartIconAdornment,
+        styles.withEndIconAdornment,
+        styles.withSearch,
+        {
+          [styles.withReversed]: reversed,
+          [styles.withDisabled]: disabled,
+          [styles.withSecondary]: secondary,
+          [styles.withLoading]: loading,
+        }
+      )}
+    >
+      <div className={styles.startIconAdornment}>
+        {loading ? (
+          <div className={styles.loadingIcon}>
+            <LoadingSpinner accessibilityLabel="Loading comments" size="sm" />
+          </div>
+        ) : (
+          <Icon icon={search} role="presentation" />
+        )}
+      </div>
+
+      <input
+        type="search"
+        value={value}
+        placeholder={placeholder}
+        disabled={disabled}
+        onChange={onChange}
+        className={classnames(
+          styles.input,
+          styles.search,
+          {
+            [styles.default]: !reversed,
+            [styles.reversed]: reversed,
+            [styles.disabled]: disabled,
+            [styles.secondary]: secondary,
+          },
+          classNameAndIHaveSpokenToDST
```

Comment:
TIL when `classNameAndIHaveSpokenToDST` is applied to primitives form components, it's applied to the `input` not the wrapper (confirmed by looking at the other primitives).

Reply:
And they're named `className` rather than `classNameAndIHaveSpokenToDST`.

What would be interesting is if we allow `classNameAndIHaveSpokenToDST` on the container (`SearchField/TextField`) level. I would imagine it should also go down to the input level to be consistent with the other native props.